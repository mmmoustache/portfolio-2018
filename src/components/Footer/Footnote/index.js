import styled from "styled-components";
import { isBrowser } from 'react-device-detect';
import { Colours } from '../../../utils/colours';
import { Timings, Durations } from '../../../utils/transitions';
import { Border as BorderSettings } from '../../../utils/layouts';
import { convertRemToPixels as rem } from '../../../utils/utils';

const Padding = `${rem('20px')}`;
const Layout = `-${rem('5px')}`;
const LinkSpacing = `${rem('40px')}`;

export default styled.div`
  padding: ${Padding};
  text-align: center;

  a {
    color: ${Colours.white};
    display: inline-block;
    font-weight: bold;
    position: relative;
    text-decoration: none;

    &::before {
      background: ${Colours.primary};
      bottom: ${Layout};
      content: '';
      display: block;
      height: ${BorderSettings.size};
      left: 0;
      position: absolute;
      transition: width ${Durations.ultrafast} ${Timings.default};
      width: 0;
      z-index: 1;
    }
    
    ${
      isBrowser && `
        &:focus,
        &:hover {
          &::before {
            width: 100%;
          }
        }
      
      `
    }

    &:last-child {
      margin-left: ${LinkSpacing};
    }
  }
`;
