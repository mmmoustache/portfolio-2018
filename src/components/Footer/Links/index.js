import styled from "styled-components";
import { isBrowser } from 'react-device-detect';
import { Colours } from '../../../utils/colours';
import { Timings, Durations } from '../../../utils/transitions';
import { convertRemToPixels as rem } from '../../../utils/utils';

const Height = `${rem('80px')}`;
const IconSize = `${rem('50px')}`;

const Link = styled.a`
  background: ${Colours.grey};
  display: block;
  height: ${Height};
  position: relative;
  transition: ${Durations.default} background ${Timings.default};
  width: 100%;

  &::before {
    background-size: contain;
    background-position: center;
    background-repeat: no-repeat;
    content: '';
    display: block;
    height: ${IconSize};
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: ${IconSize};
  }
`;

export const Github = Link.extend`
  &::before {
    background-image: url('./github.svg');
  }

  ${
    isBrowser && `
      &:focus,
      &:hover {
        background: ${Colours.gitHub};
      }    
    `
  }
`;

export const LinkedIn = Link.extend`
  &::before {
    background-image: url('./linkedin.svg');
  }

  ${
    isBrowser && `
      &:focus,
      &:hover {
        background: ${Colours.linkedIn};
      }    
    `
  }
`;

export const Links = styled.div`
  display: flex;
`;
