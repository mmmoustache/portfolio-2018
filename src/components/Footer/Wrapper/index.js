import styled from "styled-components";
import { Colours } from '../../../utils/colours';

export default styled.div`
  background: ${Colours.darkgrey};
  color: ${Colours.white};
`;
