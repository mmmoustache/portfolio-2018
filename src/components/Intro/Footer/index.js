import styled from "styled-components";
import { convertRemToPixels as rem } from '../../../utils/utils';

const offset = rem('40px');

export default styled.div`
  bottom: ${offset};
  display: flex;
  position: absolute;
  right: ${offset};
  z-index: 2;
`;
