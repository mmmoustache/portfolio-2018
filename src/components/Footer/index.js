import React from 'react';
import Footnote from './Footnote';
import Wrapper from './Wrapper';
import { Links, LinkedIn, Github } from './Links';

const Footer = () => (
  <Wrapper>
    <Links>
      <LinkedIn href="https://www.linkedin.com/in/jack-coventry-713a8811" title="View my LinkedIn Profile" target="_blank" rel="noopener noreferrer" />
      <Github href="https://github.com/mmmoustache" title="View my GitHub Profile" target="_blank" />
    </Links>
    <Footnote>
      <a href="https://github.com/mmmoustache/portfolio-2018" title="View source code on GitHub" target="_blank" rel="noopener noreferrer">Made by me</a>
      <a href="#" title="Back to top">Back to top</a>
    </Footnote>
  </Wrapper>
);

export default Footer;
