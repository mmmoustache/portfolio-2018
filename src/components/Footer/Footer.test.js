import React from 'react';
import renderer from 'react-test-renderer';
import Footer from "../Footer";

describe('Component: Footer', () => {

  it('Should render successfully', () => {
    const component = renderer.create(
      <Footer>Test</Footer>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
