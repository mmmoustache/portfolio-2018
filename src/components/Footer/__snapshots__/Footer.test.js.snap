// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Footer Should render successfully 1`] = `
<div
  className="sc-bwzfXH klcVNy"
>
  <div
    className="sc-EHOje fhxHOS"
  >
    <a
      className="sc-ifAKCX dUTKGf"
      href="https://www.linkedin.com/in/jack-coventry-713a8811"
      target="_blank"
      title="View my LinkedIn Profile"
    />
    <a
      className="sc-bxivhb ePKVXL"
      href="https://github.com/mmmoustache"
      target="_blank"
      title="View my GitHub Profile"
    />
  </div>
  <div
    className="sc-bdVaJa kYfrCF"
  >
    <a
      href="https://github.com/mmmoustache/portfolio-2018"
      target="_blank"
      title="View source code on GitHub"
    >
      Made by me
    </a>
    <a
      href="#"
      title="Back to top"
    >
      Back to top
    </a>
  </div>
</div>
`;
